src/mongocxx/CMakeLists.txt

```diff
@@ -135,78 +135,71 @@ include_directories(
     ${CMAKE_BINARY_DIR}/src
 )
 
-# We define two libraries: the normal library and the testing-only library.  The testing-only
-# library does not get installed, but the tests link against it instead of the normal library.  The
-# only difference between the two libraries is that MONGOCXX_TESTING is defined in the testing-only
-# library (this define enables special testing-only behavior).
-add_library(mongocxx
-    ${mongocxx_sources}
-)
-add_library(mongocxx_mocked
-    ${mongocxx_sources}
-)
-target_compile_definitions(mongocxx_mocked PUBLIC MONGOCXX_TESTING)
-
-if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    target_compile_options(mongocxx_mocked PRIVATE /bigobj)
-endif()
-
-set_target_properties(mongocxx_mocked PROPERTIES
-    OUTPUT_NAME mongocxx-mocked
-)
+function(mongocxx_add_library TARGET OUTPUT_NAME LINK_TYPE)
```

Comment:
Apply a similar pattern here, as I suggested with `bsoncxx_add_library` and `bsoncxx_install`.